Storage/Rust/RustFirefoxSuggestion.swift

```diff
@@ -0,0 +1,45 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import MozillaAppServices
+import UIKit
+
+/// A Firefox Suggest search suggestion. This struct is a Swiftier
+/// representation of the Rust `Suggestion` enum.
+public struct RustFirefoxSuggestion {
+    public let title: String
+    public let url: URL
+    public let isSponsored: Bool
+    public let iconImage: UIImage?
+    public let fullKeyword: String
+
+    internal init?(_ suggestion: Suggestion) {
+        // This code is intentionally written as a chain of `if-case-let`s
+        // instead of a `switch`, because we don't want new `Suggestion` cases
+        // to be a source-breaking change. A `switch` with an `@unknown
+        // default` case would emit a "default will never be executed" warning,
+        // because Swift treats `Suggestion` as frozen, since we can't build
+        // Application Services with library evolution support.
```

Comment:
So, if you use, just `default:` in a switch case (vs the `@unknown default:`) you won't get a warning. Just in case you did, in fact, want to use a switch statement here.

Reply:
Nope, `default:` without `@unknown` warns, too, unfortunately:

<img width="504" alt="Screenshot 2023-09-27 at 08 10 15" src="https://github.com/mozilla-mobile/firefox-ios/assets/41387967/2dea22e8-f5d6-4e67-a6e4-0a1fba2e37aa">

I think the `switch` is always exhaustive from Swift's perspective—it sees the enum only has two cases, and it's implicitly frozen since we aren't using library evolution—and that's why it's warning 😞 But if we used a `switch` without a default, it would be a breaking change for iOS every time we added a new suggestion type to Application Services.

Reply:
If we are unable to use `@unknown` due to build limitations then the only other way I'm aware of to silence this warning would be to add an extra case that suggests an unhandled type is acceptable, like `.none` or something along those lines. If that isn't desirable from the framework side then the `if case let `approach would be the next best option.

Reply:
> add an extra case that suggests an unhandled type is acceptable, like `.none` or something along those lines. If that isn't desirable from the framework side then the `if case let `approach would be the next best option.

That makes sense, thank you! It would be nice to avoid that on the framework side if we can. Swift is a bit of the odd one out with warning about this—enums in Kotlin (and Gecko JS) are conceptually "non-frozen"—so I'm leaning toward using `if case let` over adding a `.none` case to the framework that shouldn't be matched.